src/kfs_grpc_inference_service.cpp

```diff
@@ -80,7 +91,17 @@ ::grpc::Status KFSInferenceServiceImpl::ModelMetadata(::grpc::ServerContext* con
     std::shared_ptr<ModelInstance> instance = nullptr;
     if (!version.empty()) {
         SPDLOG_DEBUG("GetModelMetadata requested model: name {}; version {}", name, version);
-        instance = model->getModelInstanceByVersion(std::stoi(version));
+        model_version_t vers = 0;
+        try {
+            vers = std::stoll(version);
+        } catch (const std::exception& e) {
```

Comment:
We already have string conversion utilities. Maybe we could use it or add our own? https://github.com/openvinotoolkit/model_server/blob/main/src/stringutils.hpp#L133